hack/deployer/runner/gke.go

```diff
@@ -193,38 +202,60 @@ func (d *GKEDriver) setupLabelsForGCEProvider() error {
 		if storageClass.Provisioner != GoogleComputeEngineStorageProvider {
 			continue
 		}
-		// This is a GCE storage class patch it
+		// this function might be called repeatedly
+		if strings.HasPrefix(storageClass.Name, storageClassPrefix) {
+			continue
+		}
 
-		// start by removing the label that makes the storage class managed by the addon manager to prevent it from being recreated before us
-		err := exec.NewCommand(fmt.Sprintf(`kubectl label sc %s "addonmanager.kubernetes.io/mode"-`, storageClass.Name)).WithoutStreaming().Run()
+		// This is a GCE storage class copy it
+		copied := copyWithPrefixAndLabels(storageClass, labels)
+		storageClassYaml, err := yaml.Marshal(copied)
 		if err != nil {
 			return err
 		}
 
-		if storageClass.Parameters == nil {
-			storageClass.Parameters = make(map[string]string)
-		}
-		storageClass.Parameters["labels"] = labels
-		storageClassYaml, err := yaml.Marshal(storageClass)
-		if err != nil {
-			return err
+		if _, exists := existingClassNames[copied.Name]; exists {
+			// do not try to update existing classes
+			continue
 		}
 
-		if err := retry.OnError(
-			wait.Backoff{Duration: 10 * time.Millisecond, Steps: 5},
-			func(err error) bool { return true },
-			func() error {
-				return exec.NewCommand(fmt.Sprintf(`cat <<EOF | kubectl replace --force -f -
-%s
-EOF`, string(storageClassYaml))).Run()
-			},
-		); err != nil {
+		// start by removing the default storage class label so that our copy can take over that role
```

Comment:
Nit: we don't actually remove the default storage class label, but set it to false?
```suggestion
		// start by making the existing gce storage classes non-default so that our copy can support this role
```